Indicators/RogersSatchellVolatility.cs

```diff
@@ -0,0 +1,98 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the Rogers-Satchell Volatility
+    /// </summary>
+    public class RogersSatchellVolatility : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly int _period;
+        private readonly RollingWindow<TradeBar> _inputWindow;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RogersSatchellVolatility"/> class using the specified parameters
+        /// </summary> 
+        /// <param name="period">The period of moving window</param>
+        public RogersSatchellVolatility(int period)
+            : this($"RSVolat({period})", period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RogersSatchellVolatility"/> class using the specified parameters
+        /// </summary> 
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of moving window</param>
+        public RogersSatchellVolatility(string name, int period)
+            : base(name)
+        {
+            _period = period;
+            _inputWindow = new RollingWindow<TradeBar>(_period);
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= _period;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod => _period;
```

Comment:
- Please move above the constructors. 
- Don't think we actually need private period, can just forward every property to the rolling window itself?

Reply:
Fixed

Reply:
With rolling sum we don't need rolling window anymore  ... but I haven't found a Size property for Sum (like rolling window have... so I need to keep _period unless having a better idea.

Reply:
My mistake... we have _rollingSum.Window.Size